doc/library/compile/nanguardmode.txt

```diff
@@ -16,4 +16,55 @@ Guide
 
 The NanGuardMode aims to prevent the model from outputing NaNs or Infs. It has
 a number of self-checks, which can help to find out which apply node is
-generating those incorrect outputs.
+generating those incorrect outputs. It provides automatic detection on 3 types
+of abnormal values: NaNs, Infs, and abnormally big values.
+
+NanGuardMode can be used as follows:
+
+.. code-block:: python
+
+    x = T.matrix()
+    w = theano.shared(numpy.random.randn(5, 7).astype(theano.config.floatX))
+    y = T.dot(x, w)
+    fun = theano.function(
+        [x], y,
+        mode=NanGuardMode(nan_is_error=True, inf_is_error=True, big_is_error=True)
+    )
+
+While using the theano function ``fun``, it will monitor the values of each
+input and output variable of each node. As long as the abnormal values are
```

Comment:
Changed accordingly.

On Mon, Jul 6, 2015 at 5:29 PM, Mehdi Mirza <notifications@github.com>
wrote:

> In doc/library/compile/nanguardmode.txt
> <https://github.com/Theano/Theano/pull/3101#discussion_r33984785>:
>
> > +of abnormal values: NaNs, Infs, and abnormally big values.
> > +
> > +NanGuardMode can be used as follows:
> > +
> > +.. code-block:: python
> > +
> > +    x = T.matrix()
> > +    w = theano.shared(numpy.random.randn(5, 7).astype(theano.config.floatX))
> > +    y = T.dot(x, w)
> > +    fun = theano.function(
> > +        [x], y,
> > +        mode=NanGuardMode(nan_is_error=True, inf_is_error=True, big_is_error=True)
> > +    )
> > +
> > +While using the theano function ``fun``, it will monitor the values of each
> > +input and output variable of each node. As long as the abnormal values are
>
> When abnoramla values
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3101/files#r33984785>.
>



-- 
LIN, Zhouhan
Department of Computer Science and Operational Research
University of Montreal

Phone: +01-514-553-6037
Email: lin.zhouhan@gmail.com
Homepage:http://zhouhanlin.webs.com/